### PR TITLE
CPQ fields ref with context

### PR DESCRIPTION
**Pull request by adamro**

### CODE REVIEW

File: packages/salesforce-adapter/src/transformers/transformer.ts [hunk 1]
```diff
@@ -1388,21 +1388,31 @@ export const createMetadataTypeElements = async ({
...
+      [CPQ_PRICE_RULE, [
+        CPQ_LOOKUP_PRODUCT_FIELD, CPQ_LOOKUP_MESSAGE_FIELD,
+        CPQ_LOOKUP_REQUIRED_FIELD, CPQ_LOOKUP_TYPE_FIELD,
+      ]],
     ]
   )
-  return field !== undefined && path !== undefined
-    && lookUpRelativeTypes.get(path.typeName) === field.elemID.typeName
+  return field !== undefined
+    && (lookUpRelativeTypes.get(field.elemID.typeName) ?? []).includes(field.name)
```
Comment (adamro) [inline packages/salesforce-adapter/src/transformers/transformer.ts:1410]: @netama - Notice we changed the logic of how we detect relative lookUps (and probably broke the workflow ones).

File: packages/salesforce-adapter/src/filters/cpq/fields_with_context_references.ts [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+  CPQ_LOOKUP_PRODUCT_FIELD,
+  CPQ_LOOKUP_MESSAGE_FIELD,
+  CPQ_LOOKUP_REQUIRED_FIELD,
+  CPQ_LOOKUP_TYPE_FIELD,
+]
+type fieldBasedinstanceFieldsRefContext = {
+  fields: string[]
+  contextField: string
+}
+type instanceFieldsRefContext = {
```
Comment (omrilit) [inline packages/salesforce-adapter/src/filters/cpq/fields_with_context_references.ts:36]: nit: instanceFieldsRefContext => InstanceFieldsRefContext

File: packages/salesforce-adapter/src/filters/cpq/fields_with_context_references.ts [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
+const { isDefined } = values
+
+const CPQ_LOOKUP_FIELDS = [
+  CPQ_LOOKUP_PRODUCT_FIELD,
+  CPQ_LOOKUP_MESSAGE_FIELD,
+  CPQ_LOOKUP_REQUIRED_FIELD,
+  CPQ_LOOKUP_TYPE_FIELD,
+]
+type fieldBasedinstanceFieldsRefContext = {
```
Comment (omrilit) [inline packages/salesforce-adapter/src/filters/cpq/fields_with_context_references.ts:32]: nit: fieldBasedinstanceFieldsRefContext => FieldBasedinstanceFieldsRefContext

File: packages/salesforce-adapter/src/filters/cpq/fields_with_context_references.ts [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+        typeContext.contextObject
+      ))
+  })
+}
+
+const replaceFieldNamesWithFieldContextRef = (
+  instances: InstanceElement[],
+  apiNameToCustomObjects: Record<string, ObjectType>
+): void => {
+  const typeApiNameToInstances = _.groupBy(
```
Comment (omrilit) [inline packages/salesforce-adapter/src/filters/cpq/fields_with_context_references.ts:142]: nit: can be extracted to `groupByTypeApiName`

**Review by omrilit — COMMENTED** [review]
Missing UTs

**Review by omrilit — APPROVED** [review]
Approving the new code. Just make sure that it's safe to merge it regarding the workflow relative lookup

**Comment by adamro** [comment]
@omrilit - It's not :\ I need to fix it first. Thx!

**Pull request merged**